subsys/bluetooth/enocean.c

```diff
@@ -404,41 +429,16 @@ static void adv_recv(const struct bt_le_scan_recv_info *info,
 		return;
 	}
 
-	/* The data format is decided by a mix of lengths and bitfields */
-
-	if (has_shortened_name) {
-		if (len == 23 && commissioning) {
-			handle_switch_commissioning(info, buf, true);
-		}
-
-		return;
-	}
-
-	if (!(payload[8] & (BIT_MASK(3) << 5)) && len >= 12 && len <= 16) {
-		handle_switch_data(info, buf, payload);
-		return;
-	}
-
-	/* Format is ambiguous, have to rely on trial and error: */
-	struct net_buf_simple_state state;
-	int err;
-
-	if (len == 29 && commissioning) {
-		net_buf_simple_save(buf, &state);
-		err = handle_switch_commissioning(info, buf, false);
-		if (!err) {
-			/* Doing the opposite of the normal error checking
-			 * pattern on purpose here: If the message *wasn't*
-			 * switch commissioning, it could be sensor data, so we
-			 * should keep parsing.
-			 */
-			return;
-		}
-
-		net_buf_simple_restore(buf, &state);
+	switch (device_type) {
+	case DEVICE_TYPE_SENSOR:
+		handle_sensor_data(info, buf, payload, len + 1 - SIGNATURE_LEN);
```

Comment:
May be also add humidity decoding in the "handle_sensor_data()"?

Reply:
Sounds like out of scope. Please create a separate issue for this.